main.tf

```diff
@@ -429,48 +429,22 @@ resource "null_resource" "worker_pre_reqs" {
   }
 }
 
-resource "null_resource" "worker_disks" {
-  count = var.worker_count
-  depends_on = [
-    null_resource.worker_pre_reqs
-  ]
+module "storage" {
+  source = "./modules/storage"
 
-  connection {
-    type        = "ssh"
-    user        = "root"
-    private_key = chomp(tls_private_key.ssh_key_pair.private_key_pem)
-    host        = element(metal_device.worker_nodes.*.access_public_ipv4, count.index)
-  }
-
-  provisioner "file" {
-    source      = "${path.module}/templates/device_name.sh"
-    destination = "/root/bootstrap/device_name.sh"
-
-  }
-  provisioner "remote-exec" {
-    inline = [
-      "bash /root/bootstrap/device_name.sh"
-    ]
-  }
-}
-
-resource "null_resource" "install_portworx" {
   depends_on = [
     null_resource.add_kubelet_flags_to_workers,
-    null_resource.worker_disks
   ]
-  connection {
-    type        = "ssh"
-    user        = "root"
-    private_key = chomp(tls_private_key.ssh_key_pair.private_key_pem)
-    host        = metal_device.control_plane.0.access_public_ipv4
-  }
 
-  provisioner "remote-exec" {
-    inline = [
-      "VER=$(kubectl version --short | awk -Fv '/Server Version: / {print $3}')",
-      "URL='https://install.portworx.com/${var.portworx_version}?mc=false&kbver='$VER'&b=true&j=auto&kd=${urlencode("/dev/pwx_vg/pwxkvdb")}&c=${local.cluster_name}&stork=true&st=k8s&pp=IfNotPresent'",
-      "kubectl --kubeconfig /root/baremetal/bmctl-workspace/${local.cluster_name}/${local.cluster_name}-kubeconfig apply -f $URL"
-    ]
+  ssh = {
+    host             = metal_device.control_plane.0.access_public_ipv4
+    private_key      = chomp(tls_private_key.ssh_key_pair.private_key_pem)
+    user             = "root"
+    kubeconfig       = "/root/baremetal/bmctl-workspace/${local.cluster_name}/${local.cluster_name}-kubeconfig"
+    worker_addresses = metal_device.worker_nodes.*.access_public_ipv4
```

Comment:
```
# From a new / empty state:
$ terraform apply
Error: Invalid for_each argument

  on modules/portworx/main.tf line 2, in resource "null_resource" "worker_disks":
   2:   for_each = toset(var.ssh.worker_addresses)

The "for_each" value depends on resource attributes that cannot be determined
until apply, so Terraform cannot predict how many instances will be created.
To work around this, use the -target argument 
```

Reply:
replacing this line with `["fake-ip"]` will cause the provision to continue, ofcourse that won't actually work.

Reply:
swapped `for_each` out for `count` and `count.index` in ede379e, which seems to work.